net/pfSense-pkg-WireGuard/files/usr/local/pkg/wireguard/wg_service.inc

```diff
@@ -95,7 +93,7 @@ function wg_service_stop() {
 function wg_is_service_running() {
 	global $wgg;
 
-	if (!($h_lock = @fopen($wgg['pid_path'],'r')) || !file_exists($wgg['pid_path'])) {
+	if (!($h_lock = @fopen($wgg['pid_path'], 'a+')) || !file_exists($wgg['pid_path'])) {

```

Comment:
Why the mode switch here? Using `a+` will cause the file to be created if it doesn't exist, which doesn't seem necessary if it's only testing that the service is running.

Reply:
good catch, not necessary.